deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/BaseLayer.java

```diff
@@ -302,6 +316,13 @@ protected INDArray preOutput(boolean training, LayerWorkspaceMgr workspaceMgr) {
 
         INDArray ret = workspaceMgr.createUninitialized(ArrayType.ACTIVATIONS, W.dataType(), input.size(0), W.size(1));
         input.castTo(ret.dataType()).mmuli(W, ret);     //TODO Can we avoid this cast? (It sohuld be a no op if not required, however)
+
+        INDArray preNorm = ret;
+        if(hasLayerNorm()){
+            preNorm = ret.dup();
```

Comment:
Is this dup necessary?
(If it is: ````ret.dup(ret.order())``` in case input is F ordered)

Reply:
It is only necessary for backprop. Will change that accordingly.